client/src/main/java/com/chattycathy/client/model/User.java

```diff
@@ -18,9 +18,33 @@ public class User {
      * @param scanner takes input for userName
      */
     public User(Scanner scanner) {
+
+        boolean nameNonEmpty;
+        boolean validLength;
+        boolean noSpaces;
+        boolean notServer;
+        boolean validInput;
+
         do {
             log.info("Please enter a username: ");
             userName = scanner.nextLine();
-        } while (userName.isEmpty());
+
+            nameNonEmpty = !userName.isEmpty();
+            validLength = userName.length() <= 30;
+            noSpaces = !userName.contains(" ");
+            notServer = !userName.toLowerCase().contains("server");
+
+            if (!validLength) {
+                log.info("Username must be 30 or less characters");
+            }
+            if (!noSpaces) {
+                log.info("Username cannot contain spaces");
+            }
+            if (!notServer) {
+                log.info("Username cannot be 'server'");
+            }
+
+            validInput = nameNonEmpty && validLength && noSpaces && notServer;
+        } while (!validInput);
```

Comment:
Please see comment above for the addition of the function. With the function implemented this line could be adapted as follows.

```suggestion
        } while (!isValidUsername(userName));
```